### DISCUSSION TITLE
Thank You MoonDrJune  //  AssertionError

### DISCUSSION THREAD

**Discussion by ssk241002** [TechfaneTechnologies/QtsApp#10] [General]
Hey MoonDrJune

Thank you for this massive effort to help the retail community of Indian stock market.

I jut encountered an error. I posted in a below.

pl help on that .

Thanks in advance
-----

C:\Users\vishw\QtsApp>python .\example.py
Traceback (most recent call last):
File "C:\Users\vishw\QtsApp\example.py", line 9, in
from qtsapp import *
File "C:\Users\vishw\AppData\Roaming\Python\Python39\site-packages\qtsapp_init_.py", line 24, in
from qtsapp.lib import *
File "C:\Users\vishw\AppData\Roaming\Python\Python39\site-packages\qtsapp\lib.py", line 73, in
pywintypes.import_pywin32_system_module("pythoncom", globals())
File "C:\python3.9.6downloaded210821\lib\site-packages\win32\lib\pywintypes.py", line 122, in import_pywin32_system_module
assert sys.modules[modname] is not old_mod
AssertionError

**Comment by MooneDrJune** [comment]
If you are trying to run from spyder then follow this https://stackoverflow.com/a/50495714/14522034

Otherwise if you are running from cmd and the error is arising. And it's a known issue, https://github.com/mhammond/pywin32

Upgrade pywin32 by following https://github.com/mhammond/pywin32#installing-via-pip

You can install pywin32 via pip:
`pip install pywin32`

If you encounter any problems when upgrading (eg, "module not found" errors or similar), you should execute:
`python Scripts/pywin32_postinstall.py -install`

This will make some small attempts to cleanup older conflicting installs.

**Reply by BLACKWHITEHACKERSOFIICE** [reply]
Erorr

**Reply by BLACKWHITEHACKERSOFIICE** [reply]
Okey thanks

**Comment by ssk241002** [comment]
Thank you Dr.

So kind of you for taking time and replying me.

Shall implement the suggestions and will confirm you soon.

Thank you again

Senthil

**Comment by ssk241002** [comment]
Hello Dr
Good Morning
Sorry for the trouble caused to you . I encountered one more error. kindly guide me whenever you get time.

--


C:\Users\vishw\QtsApp>python .\example.py
caught a WS exception - going to restart in a few seconds
Traceback (most recent call last):
  File "C:\python3.9.6downloaded210821\lib\site-packages\qtsapp\QtsApp.py", line 107, in EnchantQtsApp
    _is_running_live, qtsapp = QtsAppRun(
  File "C:\python3.9.6downloaded210821\lib\site-packages\qtsapp\QtsApp.py", line 74, in QtsAppRun
    QTSAppStream(**kwargs)
  File "C:\python3.9.6downloaded210821\lib\site-packages\qtsapp\QTSAppStream.py", line 199, in __init__
    self._wb = xw.Book(self._wb_name)
  File "C:\python3.9.6downloaded210821\lib\site-packages\xlwings\main.py", line 887, in __init__
    for wb in app.books:
  File "C:\python3.9.6downloaded210821\lib\site-packages\xlwings\main.py", line 503, in books
    return Books(impl=self.impl.books)
  File "C:\python3.9.6downloaded210821\lib\site-packages\xlwings\_xlwindows.py", line 643, in books
    return Books(xl=self.xl.Workbooks, app=self)
  File "C:\python3.9.6downloaded210821\lib\site-packages\xlwings\_xlwindows.py", line 536, in xl
    self._xl = get_xl_app_from_hwnd(self._hwnd)
  File "C:\python3.9.6downloaded210821\lib\site-packages\xlwings\_xlwindows.py", line 308, in get_xl_app_from_hwnd
    ptr = accessible_object_from_window(child_hwnd)
  File "C:\python3.9.6downloaded210821\lib\site-packages\xlwings\_xlwindows.py", line 281, in accessible_object_from_window
    res = oledll.oleacc.AccessibleObjectFromWindow(  # noqa: F841
  File "_ctypes/callproc.c", line 997, in GetResult
OSError: [WinError -2147467259] Unspecified error

During handling of the above exception, another exception occurred:

Traceback (most recent call last):
  File "C:\Users\vishw\QtsApp\example.py", line 34, in <module>
    EnchantQtsApp(
  File "C:\python3.9.6downloaded210821\lib\site-packages\qtsapp\QtsApp.py", line 153, in EnchantQtsApp
    del qtsapp
UnboundLocalError: local variable 'qtsapp' referenced before assignment

C:\Users\vishw\QtsApp>

**Reply by MooneDrJune** [reply]
Have you followed the readme guidelines?, about updating the `.env.secret` etc... Because I don't see the output you have included stating either `session validated` or `Session Expired, Logging in again.....`...

**_Request you to kindly Follow_** [this](https://github.com/TechfaneTechnologies/QtsApp#instructions-on-running-the-program-for-the-first-time) and try again...

_And I have confirmed this, before writing this reply to you._

![image](https://user-images.githubusercontent.com/96371033/196121255-61b6d339-cdc5-4313-a8db-e837b5f6c48d.png)

![image](https://user-images.githubusercontent.com/96371033/196121332-76ecf6f8-023f-4f1f-998d-d205023b218c.png)

**Reply by ssk241002** [reply]
Dr
I am logging in thru google credentials to QtsApp. I took the API key and secret as you explained and updated the same in .env file only.

I have not updated in env.secret file.

all these I had done already and the error message sent to you was  with all the updation.

So now  I will update the API key again an see if it  works and if  I restart .

Thank you Dr

**Reply by ssk241002** [reply]
Dr
I had updated the Qtsapp API credentials again and executed the python script in CMD with admin privilege.

and with this I could successfully able to get the data on the excel.

Thank you.

if I run the script with user , it is showing some error. this is for your info only.

further , in the excel file , tha data is streamed only for Bank Nifty and no response is given for other scripts.

please guide on that please so that i can get the data for any desired scrip.

a BIG thanks to you Dr.

This is truly an amazing tool.

for your info attaching the screen shot of the streaming data

**Comment by ssk241002** [comment]
![SS of OI updation](https://user-images.githubusercontent.com/115137008/196342931-5bdc7376-e897-455c-8b8c-8b5cef6b11b3.JPG)

**Comment by mzwdnmd** [comment]
You cannot edit .xlsm files using this extension. However an exporter for macros from xlsm is in it's very early stages of development.
Xlwings created a free plugin for VSC that will allow you do edit the VBA code in your .XLSM files. I have been doing this successfully for a while now.
Check out this YouTube video that show you how to do this and where to download the install files.

https://www.youtube.com/watch?v=xoO-Fx0fTpM&ab_channel=xlwings
if I run the script with user , it is showing some error. this is for your info only.

further , in the excel file , tha data is streamed only for Bank Nifty and no response is given for other scripts.

please guide on that please so that i can get the data for any desired scrip.

a BIG thanks to you Dr.

This is truly an amazing tool.

for your info attaching the screen shot of the streaming data
